forms-flow-api/src/formsflow_api/resources/form_process_mapper.py

```diff
@@ -420,24 +399,58 @@ def delete(mapper_id: int):
     )
     def put(mapper_id: int):
         """Update form by mapper_id."""
-        application_json = request.get_json()
+        data = request.get_json()
 
-        if "taskVariable" in application_json:
+        # Extract mapper and authorization data from the request
+        mapper_data = data.get("mapper")
+        authorization_data = data.get("authorizations")
+
+        # Get the parentFormId as resource id from mapper data if authorization data is provided
+        resource_id = mapper_data.get("parentFormId") if authorization_data else None
+        task_variable = mapper_data.get("taskVariables", [])
+
+        # If task variables are present, update filter variables and serialize them
+        if "taskVariables" in mapper_data:
             FilterService.update_filter_variables(
-                application_json.get("taskVariable"), application_json.get("formId")
-            )
-            application_json["taskVariable"] = json.dumps(
-                application_json.get("taskVariable")
+                task_variable, mapper_data.get("formId")
             )
+            mapper_data["taskVariables"] = json.dumps(task_variable)
+
+        # Load the mapper data into the schema
         mapper_schema = FormProcessMapperSchema()
-        dict_data = mapper_schema.load(application_json)
+        dict_data = mapper_schema.load(mapper_data)
+
+        # Update the mapper with the provided data
         mapper = FormProcessMapperService.update_mapper(
             form_process_mapper_id=mapper_id, data=dict_data
         )
-        response = mapper_schema.dump(mapper)
-        response["taskVariable"] = json.loads(response["taskVariable"])
-        FormHistoryService.create_form_logs_without_clone(data=application_json)
 
+        # If authorization data and resource ID are provided, update resource authorization
+        if authorization_data and resource_id:
+            AuthorizationService.create_or_update_resource_authorization(
+                authorization_data, bool(auth.has_role([CREATE_DESIGNS]))
+            )
+
+        # Dump the updated mapper data into the response schema
+        mapper_response = mapper_schema.dump(mapper)
+
+        if mapper_response.get("taskVariables"):
```

Comment:
you could use walrus operator here to avoid getting same key multiple times.
if task_variables := mapper_response.get("taskVariables"):
 mapper_response["taskVariables"] = json.loads(task_variables)

Reply:
done